analyzers/tests/SonarAnalyzer.UnitTest/TestCases/DisposeFromDispose.CSharp7_2.cs

```diff
@@ -52,4 +53,99 @@ public void Dispose()
         }
     }
 
+    public class OutterClass : IDisposable
+    {
+        private Stream stream;
+
+        class InnerClass : IDisposable
+        {
+            public void M(OutterClass outter)
+            {
+                outter.stream.Dispose(); // Compliant
+            }
+
+            public void Dispose() { }
+        }
+
+        public void Dispose() { }
+    }
+
+    public class BaseClass : IDisposable
+    {
+        protected Stream stream;
+
+        public void Dispose() { }
+
+    }
+
+    public class Derived : BaseClass, IDisposable
+    {
+        public void Cleanup() => stream.Dispose(); // Compliant
+    }
```

Comment:
Would it be possible to also cover this case? I don't know how hard it would be to check whether or not the base class calls `stream.Dispose()` in its `Dispose`-method.